src/goto-programs/goto_inline.cpp

```diff
@@ -723,27 +724,27 @@ Function: goto_inline
 
   Inputs:
 
- Outputs:
+ Outputs: true, if a recursion was detected
 
- Purpose:
+ Purpose: 
 
 \*******************************************************************/
 
-void goto_inline(
+bool goto_inline(
```

Comment:
The return value seems a bit obscure; perhaps use the class directly, and then have a method recursion_detected()?